apps/design/backend/src/language_and_audio/translator.ts

```diff
@@ -116,8 +116,23 @@ export class GoogleCloudTranslator implements Translator {
     textArray: string[],
     targetLanguageCode: NonEnglishLanguageCode
   ): Promise<string[]> {
+    // Google Cloud will preserve HTML tags fairly well, so we can pass HTML
+    // rich text directly to the API. However, it has a max string length limit,
+    // so base64 encoded img src attributes are generally too long to include.
+    // We strip them out in order and replace them after translating.
+    const srcRegex = /src="([^"]*)"/g;
+    const srcPlaceholder = 'src=""';
+    const srcAttrsArray = textArray.map((text) =>
+      iter(text.matchAll(srcRegex))
+        .map((match) => match[0])
+        .toArray()
+    );
+    const textArrayWithoutSrcAttrs = textArray.map((text) =>
+      text.replaceAll(srcRegex, srcPlaceholder)
```

Comment:
Good callout! I was hoping that reordering wouldn't impact the images since tiptap is configured to only support block images, not inline. But I think adding some indexing here is a good defensive move, so I'm in favor if it's easy. I'll give it a shot

Reply:
cde84d767